From 177f05bd27c341b752cfb553abfd650782ce0381 Mon Sep 17 00:00:00 2001
From: "Sodhi, VunnyX" <vunnyx.sodhi@intel.com>
Date: Fri, 10 Apr 2015 15:11:10 +0530
Subject: [PATCH] ASoC: codec: rt5670: Headphone detection delay is reduced.

Headphone delay in detecting is reduced by
decreasing CHT_HS_DET_RETRY_COUNT to 1.

Issue: GMINL-8261

Change-Id: I4f12537a93f032792ccaa64af3a842632e47c59e
Signed-off-by: Sodhi, VunnyX <vunnyx.sodhi@intel.com>
---
 sound/soc/codecs/rt5670.c                  | 2 +-
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/rt5670.c b/sound/soc/codecs/rt5670.c
index fef391e17f6e..ff54514e98de 100644
--- a/sound/soc/codecs/rt5670.c
+++ b/sound/soc/codecs/rt5670.c
@@ -501,7 +501,7 @@ static int rt5670_readable_register(
 int rt5670_headset_detect(struct snd_soc_codec *codec, int jack_insert)
 {
 	int val;
-	int i = 0, sleep_time[5] = {10, 5, 5, 5, 5};
+	int i = 0, sleep_time[5] = {100, 5, 5, 5, 5};
 	struct rt5670_priv *rt5670 = snd_soc_codec_get_drvdata(codec);
 
 	if (jack_insert) {
diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 5d22a2d5dd14..765ac9864777 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -42,13 +42,13 @@
 #define CHT_PLAT_CLK_3_HZ	19200000
 
 #define CHT_INTR_DEBOUNCE               0
-#define CHT_HS_INSERT_DET_DELAY         500
+#define CHT_HS_INSERT_DET_DELAY         400
 #define CHT_HS_REMOVE_DET_DELAY         500
 #define CHT_BUTTON_DET_DELAY            100
 #define CHT_HS_DET_POLL_INTRVL          100
 #define CHT_BUTTON_EN_DELAY             1500
 
-#define CHT_HS_DET_RETRY_COUNT          6
+#define CHT_HS_DET_RETRY_COUNT          1
 
 struct cht_mc_private {
 	struct snd_soc_jack jack;
-- 
1.9.1

